### PR TITLE
Add an example of build compatible extension

### PR DESCRIPTION
**Pull request by manovotn**
Example of BCE with all its phases to provide a more complete picture

### CODE REVIEW

File: src/main/java/org/acme/extension/GreetingResource.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.acme.extension;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+@ApplicationScoped
```
Comment (mkouba) [inline src/main/java/org/acme/extension/GreetingResource.java]: Unnecessary annotation.

File: src/main/java/org/acme/extension/CdiExtension.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public void registrationPhase(BeanInfo beanInfo, Messages messages){
+        // browse discovered beans, inspect metadata
+        messages.info("UndiscoveredBean has scope @" + beanInfo.scope().name() + " and its qualifiers are " + beanInfo.qualifiers(), beanInfo);
+    }
+
+    @Synthesis
+    public void synthesisPhase(SyntheticComponents syntheticComponents) {
+        // allows to create synthetic beans and observers; below is a very primitive example of a String typed bean
+        syntheticComponents.addBean(String.class)
+                .type(String.class)
```
Comment (mkouba) [inline src/main/java/org/acme/extension/CdiExtension.java:47]: I believe that it's a bad practice to register a JDK type with no qualifier ;-). Maybe just use `MyQualifier` as well?

Comment (manovotn) [reply src/main/java/org/acme/extension/CdiExtension.java:47]: Fair enough, will do.

**Comment by manovotn** [comment]
@mkouba updated as per comments

**Review by mkouba — APPROVED** [review]

**Pull request merged**